src/picture.hpp

```diff
@@ -74,9 +75,9 @@ class locator
 
 	locator& operator=(const locator& a);
 
-	bool operator==(const locator& a) const { return index_ == a.index_; }
-	bool operator!=(const locator& a) const { return index_ != a.index_; }
-	bool operator<(const locator& a) const { return index_ < a.index_; }
+	bool operator==(const locator& a) const { return hash() == a.hash(); }
+	bool operator!=(const locator& a) const { return !operator==(a); }
+	bool operator<(const locator& a) const { return hash() < a.hash(); }
```

Comment:
What's the purpose of the overloaded less than operator? Comparing whether one hash is less than another hash seems like it wouldn't be especially useful?

Reply:
Hold on, this is just completely wrong. You can't implement operator== by comparing the hash, that completely defeats the point of using the locator instead of the hash as a key.

Reply:
Resolved both.